### PR TITLE
Module: use generated proptable

### PR DESCRIPTION
**Pull request by christianvuerings**
#### What changed?

* Use generated proptable for the `Module` component
* Inline `Module` flow types
* Remove `overrideTypes` (inlined)

#### Why?

Inlining the props has the following benefits:

* No more weird props spreading in order to make flow happy
* You can quickly see what the props are for a component you are working on without having to go through a habit hole of combined flow types.

### CODE REVIEW

File: docs/pages/module.js [hunk 1]
```diff
@@ -5,77 +5,18 @@ import PropTable from '../components/PropTable.js';
...
-          {
-            name: 'type',
-            href: 'static-error',
-            type: '"info" | "error"',
-            defaultValue: 'info',
-            description:
-              'If set to `error`, displays error icon and changes title to red text. Be sure to provide an `iconAccessibilityLabel` when set to `error`.',
-          },
-        ]}
+        description="A Module is a container that holds content about one subject. Its contents can be visible at all times, or expand and collapse as individual modules or a group of modules."
```
Comment (dangerismycat) [inline docs/pages/module.js:17]: Can this also be moved to the component file along with the props?

File: docs/pages/module.js [hunk 2]
```diff
@@ -582,3 +523,19 @@ function ModuleExample5() {
...
 }
+
+export async function getStaticProps(): Promise<{| props: {| generatedDocGen: DocGen |} |}> {
+  const generatedDocGen = await docgen('Module');
+
+  generatedDocGen.props.icon = {
+    ...generatedDocGen.props.icon,
+    flowType: {
+      name: 'string',
+      raw: 'Icon[icon]',
```
Comment (dangerismycat) [inline docs/pages/module.js:534]: Screen Shot 2021-12-13 at 11 14 13 AM

The current `string` type for this prop isn't helpful, but I'm not sure that this is either. Perhaps the description should include a link to `/icon` to guide the user to valid icon names?

File: docs/components/docgen.js [hunk 3]
```diff
@@ -43,16 +43,3 @@ export default async function docgen(component: string): Promise<DocGen> {
 
   return parsed;
 }
-
-export function overrideTypes(docGen: DocGen, typeOverrides: {| [string]: string |}): DocGen {
```
Comment (dangerismycat) [inline docs/components/docgen.js:47]: Given that this functionality is used in several places, is there a reason to remove this helper function and inline the logic?

File: docs/pages/module.js (hunk 1 above)
Comment (christianvuerings) [reply docs/pages/module.js:17]: @dangerismycat Definitely, updated.

File: docs/pages/module.js (hunk 2 above)
Comment (christianvuerings) [reply docs/pages/module.js:534]: @dangerismycat This uses Flow Indexed Access types: https://flow.org/en/docs/types/indexed-access/. I also added a link to https://gestalt.netlify.app/iconography_and_svgs

File: docs/components/docgen.js (hunk 3 above)
Comment (christianvuerings) [reply docs/components/docgen.js:47]: @dangerismycat I added it back in to unblock this PR. Originally I inlined it for the following reasons:

* The function has the wrong abstraction: it always assumes that the type is `union` (for module I needed `string`). We could add another prop to allow for a different type but that only makes the function more complex.
* `typeOverrides` felt like a weird format, I was expecting `{ name: 'Icon', raw: '', name: '' }`

In general I feel like we're making our code base a bit too DRY by over-abstracting code which makes it harder to see what the code is supposed to do.

More resources on this topic:
* https://www.deconstructconf.com/2019/dan-abramov-the-wet-codebase
* https://www.philosophicalhacker.com/post/when-to-dry/

**Comment by christianvuerings** [comment]
@dangerismycat I addressed all the comments - mind taking another look? Feel free to self-merge

File: docs/pages/module.js (hunk 2 above)
Comment (dangerismycat) [reply docs/pages/module.js:531]: TIL!

File: docs/components/docgen.js (hunk 3 above)
Comment (dangerismycat) [reply docs/components/docgen.js]: Totally valid reason — I think you could have left these changes in the PR. We can re-add them in a follow-up

Comment (dangerismycat) [reply docs/components/docgen.js]: Also: I get the over-abstraction pain as well. I was just glancing through the Button types the other day and was so confused. It felt like that tweet where Abramov was revisiting Redux and thought it was way too complex 😅

**Review by dangerismycat — APPROVED** [review]

**Pull request merged**